superset-frontend/packages/superset-ui-core/src/query/types/Filter.ts

```diff
@@ -58,6 +58,7 @@ export interface FreeFormAdhocFilter {
   expressionType: 'SQL';
   clause: 'WHERE' | 'HAVING';
   sqlExpression: string;
+  isExtra?: boolean;
```

Comment:
HIGHLY optional, but.. While we're at it, I believe `timeGrain` is also a valid prop for `FreeFormAdhocFilter`, similar to `BaseSimpleAdhocFilter` (at least it should be). The fact that this and `timeGrain` was missing here kinda indicates that we might benefit from some added abstraction, like

```typescript
interface BaseAdhocFilter = {
  clause: 'WHERE' | 'HAVING';
  timeGrain?: TimeGranularity;
  isExtra?: boolean;
}

type BaseSimpleAdhocFilter =  BaseAdhocFilter & {
  expressionType: 'SIMPLE';
  subject: string;
}

export type FreeFormAdhocFilter = BaseAdhocFilter & {
  expressionType: 'SQL';
  sqlExpression: string;
}
```

Reply:
Good point, done!

Reply:
Good point, done!